tests/unit/network.test.ts

```diff
@@ -99,6 +99,28 @@ describe("Network", () => {
       await expect(network.addNode("4")).rejects.toThrow("No more addresses available in 192.168.0.0/30");
     });
 
+    it("should add a node if the network was full and some node was removed", async () => {
```

Comment:
The subject looks missleading. When I reed the test, I consider "should re-use an IP address when a node previously using it leaves the network".

I'm not sure if I like the idea of re-using the IP for different nodes. Lets consider that the ID of the node is a provider or a requestor on such a network. Lets assume that providers and requetsors are "blinking" on the network - going in and out.

The typicall assumption with IPs is that you have the IP "leased" in a network for some period of time so that you retain it even if you blink shortly. Here we're trying to re-use that, which I don't think is a good idea.

- I create a network
- I put providers A, B, C in the network
- Tey get IPs: A.1, B.2, C.3
- The provider A.1 gets removed from the network
- D joins network as D.1

Won't it be the case that the providers B and C will still think that .1 means A and nod D?